### PR TITLE
Upgrade rubocop to latest version

### PR DESCRIPTION
**Pull request by dwhenry**
This is in preparation for the rails 5 upgrade

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,22 +2,22 @@ source 'https://rubygems.org'
 
 ruby IO.read('.ruby-version').strip
 
-gem 'rails', '4.2.6'
+gem 'jbuilder', '~> 2.0'
```
Comment (benlovell) [inline Gemfile]: Weird how rubocop wasn't smart enough to order the entries given a whitespace line. Shall we put them back together and order them all? It seems they've been split arbitrarily.

File: lib/response_logger.rb [hunk 2]
```diff
@@ -19,6 +19,5 @@ def call(env)
     super
   end
 
-  def on_complete(env)
-  end
+  def on_complete(env); end
```
Comment (benlovell) [inline lib/response_logger.rb]: Disable this rule please, we have in other projects also.

File: spec/features/import_booking_bug_data_spec.rb [hunk 3]
```diff
@@ -70,24 +72,24 @@
   end
 
   def given_old_booking_bug_data_exists # rubocop:disable Metrics/MethodLength
-    jun_2016 = create(:year_month, date: Time.zone.parse('2016-06-15'))
-    may_2016 = create(:year_month, date: Time.zone.parse('2016-05-15'))
+    jun2016 = create(:year_month, date: Time.zone.parse('2016-06-15'))
```
Comment (benlovell) [inline spec/features/import_booking_bug_data_spec.rb]: Was this a rubocop correction? This one's a bit weird, can't see anything wrong with the original.

Comment (dwhenry) [inline spec/features/import_booking_bug_data_spec.rb]: Apparently numbers do not signify a break in a word. I can disable this rule as well if you like.

Comment (dwhenry) [inline spec/features/import_booking_bug_data_spec.rb]: Style/VariableNumber

**Comment by dwhenry** [comment]
@benlovell all done

**Pull request merged**